src/howitz/endpoints.py

```diff
@@ -475,30 +475,83 @@ def show_update_events_status_modal():
 
+    return render_template('/responses/toggle-select.html', id=event_id, is_checked=False,
+                           is_menu=len(session["selected_events"]) > 0, show_clear_flapping=show_clear_flapping)
+
+
+@main.route('/event/<event_id>/select', methods=["POST"])
+def select_event(event_id):
+    event_type = request.form.get('eventtype')
+
+    # Store selected event id and its type in session
+    session["selected_events"][str(event_id)] = event_type
+    session.modified = True
+    current_app.logger.debug("SELECTED EVENTS %s", session["selected_events"])
+
+    selected_event_types = session["selected_events"].values()
+    if not selected_event_types:  # selected events dict is empty, should not happen
+        raise RuntimeError(f"Event {event_id} was not selected. Please try again.")
+
+    # Allow bulk clear flapping only if all of selected events are confirmed portstate events
+    show_clear_flapping = all(event_type == 'portstate' for event_type in selected_event_types)
+
+    return render_template('/responses/toggle-select.html', id=event_id, is_checked=True,
+                           is_menu=len(session["selected_events"]) > 0, show_clear_flapping=show_clear_flapping)
+
+
+@main.route('/event/bulk_clear_flapping', methods=['POST'])
+def bulk_clear_flapping():
+    selected_events = session.get("selected_events", {})
+    expanded_events = session.get("expanded_events", {})
+    current_app.logger.debug('SELECTED EVENTS %s', selected_events)
+    current_app.logger.debug('EXPANDED EVENTS %s', expanded_events)
+
+    # Update each selected event with new values
+    for event_id in selected_events:
+        flapping_res = current_app.event_manager.clear_flapping(int(event_id))
+
+        if not flapping_res:
+            raise MethodNotAllowed(description='Cant clear flapping on a non-port event.')
+
+    # Clear selected events
+    session["selected_events"] = {}
+    session.modified = True  # Necessary when modifying arrays/dicts/etc in flask session
+    current_app.logger.debug("SELECTED EVENTS %s", session["selected_events"])
+
+    # Rerender whole events table
+    poll_events_list = poll_current_events()  # Calling poll events method is needed to preserve info about which events are expanded
+    return render_template('/responses/bulk-update-events-status.html', poll_event_list=poll_events_list)
 
-@main.route('/event/<i>/select', methods=["GET"])
-def select_event(i):
-    try:
-        session["selected_events"].append(i)
-        session.modified = True
-        current_app.logger.debug("SELECTED EVENTS %s", session["selected_events"])
-    except ValueError:
-        pass
 
-    return render_template('/responses/toggle-select.html', id=i, is_checked=True,
-                           is_menu=len(session["selected_events"]) > 0)
+@main.route('/event/<i>/clear-flapping', methods=["POST"])
+def clear_flapping(i):
+    selected_events = session.get("selected_events", {})
+    event_id = int(i)
+
```

Comment:
Route: '/event/<int:event_id>/clear-flapping'

.. then you can get rid of the `event_id = int(i)` line.